### ISSUE TITLE
[Bug] Smooth World causes issues with some entities in chunkloader-loaded chunks

### ISSUE THREAD

**Issue by themrrobert** [sp614x/optifine#6369]
## Description of Issue
I've seen some posts about Smooth world already, but they aren't quite correct / accurate to this problem.

In a chunkloader-loaded chunk, most things work as they should, for example: redstone works, I have a redstone repeater with a clock and it runs just fine in the far-away chunk (thats loaded with a chunkloader). However, shulkers do not work.  I ran some other tests a while back, and I know that most entity processing works, like minecarts and dropped items from droppers, but shulkers don't work.

## Steps to Reproduce
0. Ensure Smooth world is off first.
1. Go to 1000 / x / 1000 (far outside spawn chunks) and build a simple minecart/item chunkloader. 
2. Create 2 dummy scoreboards, one for the redstone clock, and one for the shulker. *1
3. Create a redstone clock as shown in *4 pointed to a command block, with default settings, and code in *2 (place a redstone torch next to one of the redstone dusts and quickly break it to start the clock. If you leave it too long, the clock stays on and you will have to break a redstone dust to break the circuit and try again)
4. Place any block, then an observer on top of it with a string in front of it
5. Place a command block behind the observer, with default settings, and command *3
6. Places a shulker on the ground in front of the observer, so when it opens its shell, it contacts the string.
7. You should now see both counters.
8. Teleport to spawn. Both counters should continue to increase.
9. Turn on Smooth world, and you will see that the shulker counter no longer increases.

## OptiFine Version
1.17.1 HD U G9

## Installation Method
Standalone installer

## Other Installed Mods
None

## F3 Debug Screenshot
![2021-11-07_19 24 58](https://user-images.githubusercontent.com/10122432/140679585-b4d10170-7d1d-4ecd-8ab1-60e929095b1e.png)

## Prior Testing
- Does this happen in Vanilla without OptiFine? No

## Additional Information
*1:  Creating the objectives for counters.
```
/scoreboard objectives add redstoneclock dummy
/scoreboard objectives add shulker
```
*2: Redstone clock commandblock:
```
scoreboard players add dummyPlayer redstoneclock 1
```
*3: Shulker command block:
```
scoreboard players add dummyPlayer shulker 1
```
*4: Redstone clock:
![image](https://user-images.githubusercontent.com/10122432/140679110-17d3249d-5aef-4d14-a787-77af78b67744.png)

*Extra: Chunkloader (mirror on the nether side so its connected to the left-half of the portal on the nether side)
![image](https://user-images.githubusercontent.com/10122432/140679242-3e6b72c4-8ce1-4c27-b8ad-b906dea39370.png)
*Extra: Shulker counter layout. This can all be 1-block lower as per my description:
![image](https://user-images.githubusercontent.com/10122432/140679280-f4b46cb7-370c-45f6-9680-c74256532e4b.png)

**Comment by Whitelisted1** [comment]
Passive mobs, such as cows, don't move when players are more than 50 blocks away from them
I would assume this is also the case for hostile mobs, such as shulkers, so it isn't moving up and down, triggering the redstone signal

**Comment by themrrobert** [comment]
That is incorrect with regards to shulkers. They do move if the chunk is loaded. I will disable smooth world and post another screenshot from spawn so you can see. (I believe I mentioned this, but still helpful to show proof :) )
![2021-11-07_19 37 08](https://user-images.githubusercontent.com/10122432/140680500-9328a116-6403-4445-b716-ed85f62eb14a.png)


> Passive mobs, such as cows, don't move when players are more than 50 blocks away from them I would assume this is also the case for hostile mobs, such as shulkers, so it isn't moving up and down, triggering the redstone signal

**Comment by Whitelisted1** [comment]
huh, sorry if I missed that
I wouldn't know how to help then, but good luck trying to find a solution

**Comment by themrrobert** [comment]
No worries, thanks :) I just play without Smooth world, but it helps SOO much that I decided to do some extensive testing and really try and give the devs enough help to get it fixed.

> huh, sorry if I missed that I wouldn't know how to help then, but good luck trying to find a solution
